sqlmesh/dbt/model.py

```diff
@@ -399,8 +407,20 @@ def to_sqlmesh(self, context: DbtContext) -> Model:
             if physical_properties:
                 model_kwargs["physical_properties"] = physical_properties
 
-        if context.target.dialect == "snowflake" and self.snowflake_warehouse is not None:
-            model_kwargs["session_properties"] = {"warehouse": self.snowflake_warehouse}
+        if context.target.dialect == "snowflake":
+            if self.snowflake_warehouse is not None:
+                model_kwargs["session_properties"] = {"warehouse": self.snowflake_warehouse}
+
+            if self.model_materialization == Materialization.DYNAMIC_TABLE:
+                if not self.snowflake_warehouse:
+                    raise ConfigError("`snowflake_warehouse` must be set for dynamic tables")
+                if not self.target_lag:
+                    raise ConfigError("`target_lag` must be set for dynamic tables")
+
+                model_kwargs["physical_properties"] = {
```

Comment:
Don't we override here what was set on line 408?

Reply:
No? Line 408 only runs `if context.target.dialect == "bigquery"`